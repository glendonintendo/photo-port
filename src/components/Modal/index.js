import React from 'react';

function Modal({ currentPhoto }) {
	const {
		name, 
		category,
		description,
		index
	} = currentPhoto;

	return (
		<div className="modalBackdrop">
			<div className="modalContainer">
				<h3 className="modalTitle">{name}</h3>
				<img 
					alt={category}
					src={require(`../../assets/large/${category}/${index}.jpg`)?.default}
				/>
				<p>
					{description}
				</p>
				<button type="button">
					Close this modal
				</button>
			</div>
		</div>
	);
};

export default Modal;