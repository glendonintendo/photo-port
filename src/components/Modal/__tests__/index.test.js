import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();

const currentPhoto = {
	name: 'Park bench',
	category: 'landscape',
	description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
	index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
	it('renders', () => {
		render(<Modal 
			currentPhoto={currentPhoto}
			onClose={mockToggleModal}
		/>);
	})

	it('matches snapshot', () => {
		const { asFragment } = render(<Modal 
			currentPhoto={currentPhoto}
			onClose={mockToggleModal}
		/>);

		expect(asFragment()).toMatchSnapshot();
	});
});

describe('Click event', () => {
	it('calls onClose handler', () => {
		const { getByText } = render(<Modal 
			currentPhoto={currentPhoto}
			onClose={mockToggleModal}
		/>);

		fireEvent.click(getByText('Close this modal'));

		expect(mockToggleModal).toHaveBeenCalledTimes(1);
	})
})