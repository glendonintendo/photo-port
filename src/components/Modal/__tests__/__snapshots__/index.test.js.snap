// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench
      </h3>
      <img
        alt="landscape"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
